src/models/methods/subscribe.ts

```diff
@@ -0,0 +1,107 @@
+import { BaseRequest, BaseResponse } from './baseMethod'
+import { Currency, StreamType } from "../common"
+import { TransactionMetadata } from '../common/transaction'
+
+interface Book {
+    taker_gets: Currency
+    taker_pays: Currency
+    taker: string
+    snapshot?: boolean
+    both?: boolean
+}
+  
+export interface SubscribeRequest extends BaseRequest {
+    command: "subscribe"
+    streams?: StreamType[]
+    accounts?: string[]
+    accounts_proposed?: string[]
+    books?: Book[]
+    url?: string
+    url_username?: string
+    url_password?: string
+}
+
+export interface SubscribeResponse extends BaseResponse {
+    result: any
+}
+
+interface BaseStream {
+    type: string
+}
+
+export interface LedgerStream extends BaseStream {
+    type: "ledgerClosed"
+    fee_base: number
+    fee_ref: number
+    ledger_hash: string
+    ledger_index: number
+    ledger_time: number
+    reserve_base: number
+    reserve_inc: number
+    txn_count: number
+    validated_ledgers?: string
+}
+
+export interface ValidationStream extends BaseStream {
+    type: "validationReceived"
+    amendments?: string[]
+    base_fee?: number
+    flags: number
+    full: boolean
+    ledger_hash: string
+    ledger_index: string
```

Comment:
This one can be string or integer, so probably should be LedgerIndex type
(https://xrpl.org/subscribe.html#validations-stream)

Reply:
No, `String - Integer` means the object is a string, but is essentially an integer cast as a string (see the example). It's to avoid floating point/overflow errors.